protocols/kad/src/behaviour.rs

```diff
@@ -2471,6 +2547,25 @@ pub enum KademliaEvent {
     PendingRoutablePeer { peer: PeerId, address: Multiaddr },
 }
 
+/// Information about progress events.
+#[derive(Debug, Clone)]
+pub struct ProgressStep {
+    /// The index into the event
+    pub count: usize,
```

Comment:
```suggestion
    pub count: NonZeroUsize,
```

To avoid confusion on whether this is zero-indexed or not, how about using a `NonZeroUsize` here?

Reply:
done